tests/test_decorator_no_args.py

```diff
@@ -1,51 +1,35 @@
-from datetime import datetime
-from typing import Tuple
-from unittest.mock import MagicMock
+from typing import List, Tuple
 
 import pytest
-from pytest_mock import MockerFixture
+from boto3 import Session
+from mypy_boto3_organizations.type_defs import AccountTypeDef
 
 from botocove import CoveSession, cove
 
 
 @pytest.fixture()
-def patch_boto3_client(mocker: MockerFixture) -> MagicMock:
-    mock_boto3 = mocker.patch("botocove.cove_host_account.boto3")
-    list_accounts_result = {"Accounts": [{"Id": "12345689012", "Status": "ACTIVE"}]}
-    mock_boto3.client.return_value.get_paginator.return_value.paginate.return_value.build_full_result.return_value = (  # noqa E501
-        list_accounts_result
-    )
-    describe_account_results = [
-        {
-            "Account": {
-                "Id": "1234",
-                "Arn": "hello-arn",
-                "Email": "email@address.com",
-                "Name": "an-account-name",
-                "Status": "ACTIVE",
-                "JoinedMethod": "CREATED",
-                "JoinedTimestamp": datetime(2015, 1, 1),
-            }
-        }
-    ]
-    mock_boto3.client.return_value.describe_account.side_effect = (
-        describe_account_results
-    )
-    return mock_boto3
+def org_accounts(mock_session: Session) -> List[AccountTypeDef]:
+    """Returns a list of the accounts in the mock org. Index 0 is the management
+    account."""
+    org = mock_session.client("organizations")
+    org.create_organization(FeatureSet="ALL")
+    org.create_account(Email="email@address.com", AccountName="an-account-name")
+    return org.list_accounts()["Accounts"]
 
 
-def test_decorated_simple_func(patch_boto3_client: MagicMock) -> None:
+@pytest.mark.usefixtures("mock_session")
```

Comment:
Is there a functional difference between mark.usefixtures() and passing them as a func arg? I feel like the inconsistant signature of the tests is making me itchy :)

Reply:
My interpretation of the Pytest documentation is that `usefixtures` is preferred when you need a fixture for its side effects but you don't access it directly by name.

> Sometimes test functions do not directly need access to a fixture object. For example, tests may require to operate with an empty directory as the current working directory but otherwise do not care for the concrete directory.

That and it avoids a soft nagging from my editor :-) VS Code renders an unused function parameter in a different color as if it were dead code.

I also did find it strange at first. If you prefer to keep specifying all fixtures via function parameters I don't mind changing it. Then the color coding becomes a sort of documentation :-D

Reply:
It's a thin hair to split for sure - https://docs.pytest.org/en/6.2.x/fixture.html#use-fixtures-in-classes-and-modules-with-usefixtures to me reads a bit like usefixtures is for 1 -> n relationships in the general case, although it does fix the IDE experience. My first reaction without a great deal of consideration is that given we're type hinting the unused fixture arg, it's worth the unused graying out: I like your documentation-as-unused idea, plus it's clickable for seeing the implementation unlike the string.

Unless you're keen to keep it, I vote sticking to passing args to trigger fixtures :)

Reply:
I vote the same. I won't be able to make any more changes now till next week so feel free to commit on top of this if you want to get it done before then.

Reply:
I've removed all the `usefixtures` decorators and replaced them with parameters where necessary for the side effects.